### PR TITLE
fix: fetch gsheet project id from cs added

### PR DESCRIPTION
**Pull request by sneha122**
## Description

File picker implementation for Limiting Gsheet Access requires google sheet project id. The changes in this PR gets the gsheet project id from cloud-services and returns it back to client. Client then uses this project id to open file picker and select required files.

Fixes https://github.com/appsmithorg/appsmith/issues/21298, https://github.com/appsmithorg/appsmith/issues/21362

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/AuthenticationResponse.java [hunk 1]
```diff
@@ -28,4 +28,6 @@ public class AuthenticationResponse implements AppsmithDomain {
 
     @Encrypted
     Object tokenResponse;
+
+    String projectID;
```
Comment (sumitsum) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/AuthenticationResponse.java:32]: Is `projectID` something that we want to store as encrypted on the DB ? If so, we should use @Encrypted annotation for it.

**Review by sumitsum — CHANGES_REQUESTED** [review]

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/AuthenticationResponse.java (hunk 1 above)
Comment (sneha122) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/AuthenticationResponse.java:32]: there is no need to store projectID on DB as we are fetching it on demand and its the same for all datasources, so should I add Transient annotation instead so we do not save it in DB at all? Also in that case @encrypted is required?

Comment (sumitsum) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/AuthenticationResponse.java:32]: Agree, let us mark it as @Transient

Comment (sumitsum) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/AuthenticationResponse.java:32]: Also, can you please mention a comment here saying that this field is not something that is returned as a response from the authorization server but it is something that the cloud service is providing.

**Review by sumitsum — CHANGES_REQUESTED** [review]

**Comment by sneha122** [comment]
`Also, should we rename it to projectIDPrefix as I am assuming it is not the full project id but only the prefix part of it ?` - Project ID is the prefix part of client Id, essentially it already is a prefix @sumitsum

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/AuthenticationResponse.java (hunk 1 above)
Comment (sneha122) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/AuthenticationResponse.java:32]: ProjectID is prefix part of clientId so it already is a prefix.

**Comment by sumitsum** [comment]
Server side changes LGTM.

**Review by Irongade — APPROVED** [review]
Generally, I'd like to encourage you to add more comments in your code. Especially for new features, this will help you and anyone else looking into this in the future.

**Comment by sneha122** [comment]
> Generally, I'd like to encourage you to add more comments in your code. Especially for new features, this will help you and anyone else looking into this in the future.

Thanks @Irongade for the feedback!! Usually I add comments in the hard to understand areas, but is there any guidelines that you follow for comments?

**Review by sumitsum — DISMISSED** [review]

**Comment by sneha122** [comment]
The code review is done for this but keeping it in code review pipeline because in order to move this to needs QA, cs PR needs to be merged, and code review for that is pending as of now

**Review by Irongade — DISMISSED** [review]

**Review by sumitsum — APPROVED** [review]

**Comment by sneha122** [comment]
This PR cannot be tested on DP, as DP generated with `build-deploy-preview` does not have cloud-services configured correctly. Hence as per this [thread](https://theappsmith.slack.com/archives/C04Q58L86RW/p1679318148732489), we have decided to merge the PR and test the changes on release. CC: @prapullac @rohan-arthur

**Comment by prapullac** [comment]
As per above comment adding the label.

**Pull request merged**